programs/bpf_loader/src/syscalls/cpi.rs

```diff
@@ -1167,13 +1375,33 @@ fn update_caller_account(
             );
             return Err(Box::new(InstructionError::InvalidRealloc));
         }
+
+        // If the account has been shrunk, we're going to zero the unused memory
+        // *that was previously used*.
         if post_len < prev_len {
```

Comment:
nit: imo, `if prev_len > post_len` reads better maybe due to less cognitive load.

Reply:
Funny, I personally prefer it the other way around and use `a < b < c` as it is usually done in mathematical notation. Also matches the notion of a X-axis flowing from left / small to right / big.

Reply:
hehe, i'm open for others perceptions.

yeah, i like usually prefer `<`. but i have stronger precedence operand ordering rule of `if old [<>] new` for comparing state changes of the same thing.